cfme/tests/networks/test_sdn_crud.py

```diff
@@ -41,3 +43,25 @@ def test_sdn_crud(provider, appliance):
     provider.wait_for_delete()
 
     assert not network_provider.exists
+
+
+@pytest.mark.ignore_stream('5.9')
+@pytest.mark.provider([OpenStackProvider], scope='function', override=True)
+def test_router_crud(provider, appliance):
+    router_name = 'test_router_crud-{}'.format(fauxfactory.gen_alphanumeric().lower())
+    routers = appliance.collections.network_routers
+    router = routers.create(name=router_name,
+                            provider=provider,
+                            tenant=provider.data.networks.cloud_tenant,
+                            network_manager=provider.data.networks.network_manager)
+    assert router.exists
+
+    router.delete()
+    provider.refresh_provider_relationships()
+
+    def router_doesnt_exists():
+        # Navigate somewhere else then to Router Details to enforce page refresh.
```

Comment:
@jaryn Why not use the exists ```exists``` method: https://github.com/ManageIQ/integration_tests/blob/master/cfme/networks/network_router.py#L74

Reply:
Because it is not refreshing the page.

Reply:
condensing conversation to L67